client/my-sites/customize/main.jsx

```diff
@@ -69,8 +70,13 @@ var Customize = React.createClass( {
 	},
 
 	redirectIfNeeded: function( menusUrl, site ) {
+		if ( isRedirecting ) {
+			return;
+		}
 		if ( site && menusUrl !== '/customize/menus/' + site.slug ) {
```

Comment:
Oh, may bad. I was trying to exclude the chance of redirect loops with the check here, but forgot that for unverified e-mail, we really redirect to `'/customize' + site.slug`.

Anyway, the `isRedirecting` flag is probably fine, tho I think you if you decide to use it, you could remove the other check. Alternatively, I guess you could change this line to

```js
if ( site && includes( [ '/customize/menus/' + site.slug, '/customize/' + site.slug ], menusUrl ) ) {
```

(using lodash's `includes`).